src/updater/formatPlugin.ts

```diff
@@ -16,14 +16,21 @@ export interface FormatPlugin<Content> {
   update(
     actual: Content | null,
     updater: Updater<Content>,
-    options: FormatPluginFnOptions
+    options: FormatPluginFnOptions,
   ): PromiseOrValue<Content | null>
 
   /** Called only if check for equality is required (`actual != null` & `expected != null`) */
   equal(expected: Content, actual: Content, options: FormatPluginFnOptions): PromiseOrValue<boolean>
 
   /** Called only if write is required (`--test` isn't specified, `expected != null` and `expected` is not equal to `actual`) */
   write(expected: Content, options: FormatPluginFnOptions): PromiseOrValue<void>
+
+  /**
+   * Used to clone the object returned by `read` before passing it to `update`.
+   * Defaults to `structuredClone`[1] if available, otherwise `v8.deserialize(v8.serialize(obj))`.
+   * [1]: https://developer.mozilla.org/en-US/docs/web/api/structuredclone
+   */
+  clone?(value: Content, options: FormatPluginFnOptions): Content
```

Comment:
One caveat here is that, when a user implements the interface, TS sometimes tries to use `.clone()`'s `value` argument to infer the type of `Content`, which is likely not what the user wants.

The upcoming `NoInfer` utility type in TypeScript 5.4 would come in handy here. In lieu of that, I've tested [this workaround from StackOverflow](https://stackoverflow.com/a/56688073/947478), which solves the issue:

```ts
type NoInfer<T> = [T][T extends any ? 0 : never]

export interface FormatPlugin<Content> {
  // ... snip ...
  clone?(value: NoInfer<Content>, options: FormatPluginFnOptions): NoInfer<Content>
}
```

I'm not sure whether to include this or not. Happy to hear your thoughts.